test/mainnet/matic.test.ts

```diff
@@ -0,0 +1,199 @@
+        //     const signerPubKey = wallet.publicKey
+        //     newMaticSigners.push(wallet.address)
+        //     const validatorSigner = await ethers.provider.getSigner(validator.signer)
+        //     await MaticStakeManager.connect(validatorSigner).updateSigner(i, signerPubKey)
+        //     await hre.network.provider.request({
+        //         method: "hardhat_stopImpersonatingAccount",
+        //         params: [validator.signer],
+        //     });
+        // }
+    })
+
+    before('deploy Tenderizer', async () => {
+        process.env.NAME = 'Matic'
+        process.env.SYMBOL = 'MATIC'
+        process.env.CONTRACT = MATIC_STAKE_MANAGER
+        process.env.VALIDATOR = VALIDATOR_SHARE
+        process.env.TOKEN = MATIC_TOKEN
+        process.env.STEAK_AMOUNT = STEAK_AMOUNT
+        process.env.ADMIN_FEE = '0'
+        process.env.SWAP_FEE = '5000000'
+        process.env.AMPLIFIER = '85'
+
+        // Deploy fixture
+        // Matic = await hre.deployments.fixture(['Matic'], {
+        //     keepExistingDeployments: true
+        // })
+
+        // comment out when 'keepExistingDeployments = false'
+
+        Tenderizer = (await ethers.getContractAt('Matic', '0xe07c344cB6a2Af8Fdf1d64c67D4C33a133fE7289')) as Matic
+        ValidatorShare = (await ethers.getContractAt('IMatic', await Tenderizer.node())) as IMatic
+    })
+
+    describe('Deploy upgrade to fix withdraw Locks', async () => {
+        before(async () => {
+            await hre.network.provider.request({
+                method: 'hardhat_impersonateAccount',
+                params: [TENDERIZE_OWNER]
+            })
+            await signers[0].sendTransaction({
+                to: TENDERIZE_OWNER,
+                value: ethers.utils.parseEther("1")
+            })
+            const deployerSigner = ethers.provider.getSigner(TENDERIZE_OWNER)
+            const newMatic = await (await ethers.getContractFactory('Matic', deployerSigner)).deploy()
+            const proxy = (await ethers.getContractAt('EIP173Proxy', Tenderizer.address)) as EIP173Proxy
+            await proxy.connect(deployerSigner).upgradeToAndCall(newMatic.address, Tenderizer.interface.encodeFunctionData('setWithdrawLockStart', [2]))
+            await hre.network.provider.request({
+                method: 'hardhat_stopImpersonatingAccount',
+                params: [TENDERIZE_OWNER]
+            })
+            await MaticToken.approve(Tenderizer.address, ethers.utils.parseEther(STEAK_AMOUNT))
+            await Tenderizer.deposit(ethers.utils.parseEther(STEAK_AMOUNT), { gasLimit: 500000 })
+            await Tenderizer.claimRewards()
+
+
+        })
+
+        it('matches withdrawal lock ids for Tenderize and Matic', async () => {
+            const tenderizeID = await Tenderizer.callStatic.unstake(ethers.utils.parseEther(STEAK_AMOUNT).div(2))
+            await Tenderizer.unstake(ethers.utils.parseEther(STEAK_AMOUNT).div(2))
```

Comment:
why do `callStatic` first and after normal call?

Reply:
To get the withdrawLockID for that WithdrawLock from the POV of Tenderizer without having to fetch it from the event. 

The function returns the withdrawLockID, but since it's not a `view` function JS libraries can't handle the return value.